PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,169 @@
+import UIKit
+#elseif os(watchOS)
+import UIKit
+import WatchKit
+#elseif os(macOS)
+import AppKit
+#endif
+
+@objc(RCSystemInfo) open class SystemInfo: NSObject {
+
+    #if targetEnvironment(macCatalyst)
+    static let platformHeaderConstant = "uikitformac"
+    #elseif os(iOS)
+    static let platformHeaderConstant = "iOS"
+    #elseif os(watchOS)
+    static let platformHeaderConstant = "watchOS"
+    #elseif os(tvOS)
+    static let platformHeaderConstant = "tvOS"
+    #elseif os(macOS)
+    static let platformHeaderConstant = "macOS"
+    #endif
+
+    public enum SystemInfoError: Error {
+        case invalidInitializationData
+    }
+
+    private static let defaultServerHostName = "https://api.revenuecat.com"
+
+    @objc public var finishTransactions: Bool
+    @objc public let platformFlavor: String
+    @objc public let platformFlavorVersion: String?
+    @objc public static var forceUniversalAppStore: Bool = false
+    @objc public static var isSandbox: Bool {
+        let url = Bundle.main.appStoreReceiptURL
+        guard let url = url else {
+            return false
+        }
+
+        let receiptURLString = url.path
+        return receiptURLString.contains("sandboxReceipt")
+    }
+    @objc public static var frameworkVersion: String { // TODO: automate the setting of this, if it hasn't been.
+        return "3.12.0-SNAPSHOT"
+    }
+
+    @objc public static var systemVersion: String {
+        return ProcessInfo().operatingSystemVersionString
+    }
+
+    @objc public static var appVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    @objc public static var buildVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static var platformHeader: String {
+        return Self.forceUniversalAppStore ? "iOS" : platformHeaderConstant
+    }
+
+    @objc public static var identifierForVendor: String? {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+            return UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var privateProxyURL: URL?
+    @objc public static var proxyURL: URL? {
+        get {
+            return privateProxyURL
+        }
+        set {
+            privateProxyURL = newValue
+            if let privateProxyURLString = newValue?.absoluteString {
+                Logger.info(Strings.configure.configuring_purchases_proxy_url_set
+                                .replacingOccurrences(of: "%@", with: privateProxyURLString))
+            }
+        }
+    }
+
+    static var defaultServerHostURL: URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static var serverHostURL: URL {
```

Comment:
I think we should try to keep the public stuff at the top, so the file is easier to read for anyone who's not looking to modify it

Reply:
I agree, with a small caveat that I group private vars that back public vars right above the public get/set declaration.